addons/stock/report/report_stock_reception.py

```diff
@@ -0,0 +1,273 @@
+                    continue
+                source = out_move.picking_id if out_move.picking_id else out_move._get_source_document()
+                if not source:
+                    continue
+                qty_assigned = min(total_assigned, out_move.product_qty)
+                sources_to_lines[source].append(
+                    self._prepare_report_line(qty_assigned, product_id, out_move, source, is_assigned=True, move_ins=moves_in))
+
+        # dates aren't auto-formatted when printed in report :(
+        sources_to_formatted_scheduled_date = defaultdict(list)
+        for source, dummy in sources_to_lines.items():
+            sources_to_formatted_scheduled_date[source] = self._get_formatted_scheduled_date(source)
+
+        return {
+            'data': data,
+            'doc_ids': docids,
+            'doc_model': 'stock.picking',
+            'sources_to_lines': sources_to_lines,
+            'precision': self.env['decimal.precision'].precision_get('Product Unit of Measure'),
+            'pickings': pickings,
+            'sources_to_formatted_scheduled_date': sources_to_formatted_scheduled_date,
+        }
+
+    def _prepare_report_line(self, quantity, product, move_out, source=False, is_assigned=False, is_qty_assignable=True, move_ins=False):
+        return {
+            'source': source,
+            'product': {
+                'id': product.id,
+                'display_name': product.display_name
+            },
+            'uom': product.uom_id.display_name,
+            'quantity': quantity,
+            'is_qty_assignable': is_qty_assignable,
+            'move_out': move_out,
+            'is_assigned': is_assigned,
+            'move_ins': move_ins and move_ins.ids or False,
+        }
+
+    def _get_formatted_scheduled_date(self, source):
+        """ Unfortunately different source record types have different field names for their "Scheduled Date"
+        Therefore an extendable method is needed.
+        """
+        if source._name == 'stock.picking':
+            return format_date(self.env, source.scheduled_date)
+        return False
+
+    def action_assign(self, move_ids, qtys, in_ids):
+        """ Links picking move(s) to the next step (i.e. move_ids in another picking)
+        """
+        for out, qty_to_link, ins in zip(move_ids, qtys, in_ids):
+            out = self.env['stock.move'].browse(out)
+            potential_ins = self.env['stock.move'].browse(ins)
+
+            if float_compare(out.product_qty, qty_to_link, precision_rounding=out.product_id.uom_id.rounding) == 1:
+                # we are only assigning part of its demand => let's split it now to prevent reservation problems later
+                # new move = the qty we're not assigning (and won't be visible in report)
+                new_move_vals = out._split(out.product_qty - qty_to_link)
+                new_out = self.env['stock.move'].create(new_move_vals)
```

Comment:
Possible to create and confirm outside the for loop ?

Reply:
Added via a fixup. It's a bit ugly looking though due to needing to match the created move with the original move later on... Also, `.create()` will always return the records in the same order as the list of values to create, correct?